model_databank/mixins.py

```diff
@@ -0,0 +1,90 @@
+# (c) Nelen & Schuurmans.  GPL licensed, see LICENSE.rst.
+# -*- coding: utf-8 -*-
+import logging
+
+from django.conf import settings
+from django.contrib.auth.mixins import AccessMixin
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.translation import ugettext as _
+
+import requests
+import six
+
+logger = logging.getLogger(__name__)
+
+
+class RoleRequiredRemoteMixin(AccessMixin):
+    """
+    Subclass of AccessMixin that fetches permission info from a remote 3Di
+    permissions API.
+
+    """
+    raise_exception = True
+    permission_denied_message = _(
+        "You're not allowed to see this page. Contact your administrator to "
+        "request access permission.")
+    role_required = None  # can be a string or a list of strings
+
+    def get_role_required(self):
+        """
+        Override this method to override the role attribute.
+        Must return an iterable.
+        """
+        if not self.role_required:
+            raise ImproperlyConfigured(
+                '{0} is missing the role attribute. Define {0}.role, or '
+                'override {0}.get_role_required().'.format(
+                    self.__class__.__name__)
+            )
+        if isinstance(self.role_required, six.string_types):
+            roles = (self.role_required,)
+        else:
```

Comment:
Is dit wel nodig? Deze mixin wordt maar op een plek gebruikt tot nu toe (ModelUploadFormView), waar role_required gewoon een string is.